src/document-click.js

```diff
@@ -358,3 +347,52 @@ export function expandTargetHref_(e, target, urlReplacements) {
   }
   return newHref;
 }
+
+
+/** @typedef {{element: !Element, attributes: Object<string, string>}} */
+let RestoreElementAttributesDef;
+
+
+/**
+ * Removes attributes from elements with matching hash target. This includes:
+ *   *[id=hash] and a[name=hash].
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {string} hash
+ * @return {!Array<!RestoreElementAttributesDef>}}
+ * @private
+ */
+function removeAttrsWithMatchingHash_(ampdoc, hash) {
+  const restoreElementsAttrs = [];
+  const targetElements = ampdoc.getRootNode().querySelectorAll(
+      `#${hash},a[name=${hash}]`) || [];
+  for (let i = 0; i < targetElements.length; i++) {
+    const element = targetElements[i];
+    const attributes = {};
```

Comment:
I think he's saying the `for attr in attrs` [loop](https://github.com/ampproject/amphtml/pull/5381/files#diff-db0430e49cbdcc5a4a002c5bbd3567b9R399) is too much. We only care about `name` and `id`, nothing else.